### PR TITLE
Go mod support

### LINKED ISSUE #5: Go modules
In order to make your project GOPATh independent, you should use `go modules`.

At the root of your project

```
go mod init github.com/fallais/gocoop
```

And then globally replace all `gocoop/*` by `github.com/fallais/gocoop/`.

It allows you to handle dependencies directly bound to your project, and prevent GOPATH poisoning problem.

I can make a PR if you want.

### PR DESCRIPTION
**Pull request by Zenithar**
# Context

In order to deploy go module support in your project, I made some changes:

- [x] Initialize a go module
- [x] Do some package relocations
- [x] Make it build on macosx

# References

- closes #5

### CODE REVIEW

File: pkg/coop/coop.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ const DefaultLongitude = 1.4437
 type Coop struct {
 	openingCondition conditions.Condition
 	closingCondition conditions.Condition
-	door             *door.Door
+	door             door.Door
```
Comment (Zenithar) [inline pkg/coop/coop.go:42]: Use the exposed interface not the platform dependent struct.

File: pkg/coop/options.go [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+	"github.com/fallais/gocoop/pkg/coop/conditions"
+	"github.com/fallais/gocoop/pkg/door"
 )
 
 // Options are the options for the coop.
 type Options struct {
 	OpeningCondition conditions.Condition
 	ClosingCondition conditions.Condition
-	Door             *door.Door
+	Door             door.Door
```
Comment (Zenithar) [inline pkg/coop/options.go:12]: Use the exposed interface not the platform dependent struct.

File: pkg/door/door.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package door
+
+// Door operation contract.
```
Comment (Zenithar) [inline pkg/door/door.go:3]: Abstract door operations as an interface it's more easy to create tests (mock) and materialize the platform-dependent boundary.

**Comment by fallais** [comment]
Thanks a lot @Zenithar !
